unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/BackpackEditorHUDController.cs

```diff
@@ -82,10 +87,56 @@ public class BackpackEditorHUDController
             view.OnColorChanged += OnWearableColorChanged;
             view.OnColorPickerToggle += OnColorPickerToggled;
             view.OnAvatarUpdated += OnAvatarUpdated;
+            view.OnOutfitsOpened += OnOutfitsOpened;
+
+            outfitsController.OnOutfitEquipped += OnOutfitEquipped;
 
             SetVisibility(dataStore.HUDs.avatarEditorVisible.Get(), saveAvatar: false);
         }
 
+        private void OnOutfitEquipped(OutfitItem outfit)
+        {
+            Dictionary<string, WearableItem> keyValuePairs = new Dictionary<string, WearableItem>(model.wearables);
+            foreach (KeyValuePair<string, WearableItem> keyValuePair in keyValuePairs)
+                UnEquipWearable(keyValuePair.Key, UnequipWearableSource.None, false, false);
+
+            foreach (string forcedCategory in model.forceRender)
+                UpdateOverrideHides(forcedCategory, false, false);
+
+            outfitLoadCancellationToken = new CancellationTokenSource();
+            LoadAndEquipOutfitWearables(outfit, outfitLoadCancellationToken.Token).Forget();
+        }
+
+        private async UniTaskVoid LoadAndEquipOutfitWearables(OutfitItem outfit, CancellationToken cancellationToken)
+        {
+            if (!wearablesCatalogService.WearablesCatalog.ContainsKey(outfit.outfit.bodyShape))
+                await wearablesCatalogService.RequestWearableAsync(outfit.outfit.bodyShape, cancellationToken);
```

Comment:
i see a lack of exception handling.. i think you should be able to equip the rest of the wearables if one fails.. dont forget to properly do the properly logging so we can detect it